documentation/managing_modules.md

```diff
@@ -0,0 +1,183 @@
+
+After you've selected your project's direct dependencies, Bolt adds those
+modules to your `bolt-project.yaml` file and pins them to their current
+versions, installs those modules and their dependencies into a `.modules`
+directory, and generates a Puppetfile. 
+
+If you don't want your modules pinned to a specific version, edit out the
+`version_requirement` fields for each module in your `bolt-project.yaml` file
+and run `bolt module install`.
+
+> **NOTE:** Migrating a project makes irreversible changes to the project's
+  configuration and inventory files. Before you migrate, make sure the project
+  has a backup or uses a version control system.
+
+## Install modules in a Bolt project
+
+Once you've opted in to the experimental workflows, you can use the `bolt module
+add` and `bolt module install` commands to manage your Forge modules. Git
+modules are not currently supported.
+
+### Add a Forge module and its dependencies to an existing project
+
+To add a single module and its dependencies to your Bolt project, use the 
+`bolt module add` command. For example, to add the puppetlabs/apt module:
+
+*\*nix shell command*
+```shell
+bolt module add puppetlabs/apt
+```
+
+*Powershell cmdlet*
+```shell
+Add-BoltModule -Module puppetlabs/apt
+```
+
+Bolt adds a declaration to your project configuration file, resolves
+dependencies, updates your project's Puppetfile, and installs the modules to the
+project's `.modules` directory.
+
+### Install the modules associated with a Bolt project
+
+If you've just downloaded a Bolt project from source control and want to install
+the modules associated with it, use the following command:
+
+*\*nix shell command*
+```shell
+bolt module install
+```
+
+*Powershell cmdlet*
+```shell
+Install-BoltModule
+```
+
+Bolt loads your `bolt-project.yaml` file, resolves dependencies for all of the
+declared modules, writes a Puppetfile to your project that includes
+specifications for the resolved modules, and installs the modules to the
+project's `.modules` directory.
+
+## How Bolt resolves shared dependencies
+
+Each time you add a new module to a project, Bolt attempts to resolve all of the
+modules in the project configuration file to ensure that there are no version
+conflicts between modules and their dependencies.
+
+If a new module shares a dependency with another module, Bolt might need to
+install a different version of the dependency that is compatible with both
+modules. When adding a new module, Bolt attempts to change the version of as few
+modules as possible. Bolt resolves modules in the following order:
+1. Pin all installed modules to a specific version and resolve with the new
+   module. Does not modify the version of any installed module.
+2. Unpin all installed modules and resolve with the new module. Update all
+   modules to the latest compatible version.
+
+If Bolt successfully resolves modules at any step, it does not continue to the
+next step.
+
+## Forcibly installing modules
```

Comment:
Likewise, this only applies to `bolt module install`, so maybe having it rolled into that section?